src/search/parser/lexical_analyzer.cc

```diff
@@ -29,6 +29,7 @@ static vector<pair<TokenType, regex>> construct_token_type_expressions() {
         {TokenType::INTEGER,
          R"([+-]?(infinity|\d+([kmg]\b)?))"},
         {TokenType::BOOLEAN, R"(true|false)"},
+        {TokenType::STRING, R"("(\\\\|\\"|\\n|[^"\\])*")"},
```

Comment:
The order here is important: for example, we have to parse identifiers after booleans because otherwise `true` would be treated as an identifier. Previously, I think we also had to parse floats before ints but as far as I can tell, all regexes except for the one of identifiers are non-overlapping now (for example, floats have to contain either `.`, or `e`, and ints may not contain either).

I ordered the tokens in a consistent way in the places I could find.

Reply:
Actually, this was wrong: parsing `1.2` if we parse integers first, would parse it as an integer token `1` followed by a float token `.2`. I switched the order again and added comments to explain why the order is what it is.

Reply:
i like the added comments.
However, now the order in the files you adjusted in https://github.com/aibasel/downward/pull/170/commits/8d4277d5df0f3d53b96710afdd400a47e33356ae do not match anymore

Reply:
Yes, but now it is clear why it doesn't match. I wouldn't change the order everywhere to match the parse order.

Reply:
that's fine for me